File: mobile/lib/core/theme/joi_typography.dart
import 'package:flutter/material.dart';

abstract final class JoiTypography {
  static const fontFamily = 'Inter';
  static const fontFamilyMono = 'JetBrains Mono';

  static const headlineLarge = TextStyle(
    fontSize: 28,
    fontWeight: FontWeight.w600,
    height: 1.3,
    fontFamily: fontFamily,
  );
  static const headlineMedium = TextStyle(
    fontSize: 22,
    fontWeight: FontWeight.w600,
    height: 1.3,
    fontFamily: fontFamily,
  );
  static const headlineSmall = TextStyle(
    fontSize: 18,
    fontWeight: FontWeight.w600,
    height: 1.4,
    fontFamily: fontFamily,
  );

  static const bodyLarge = TextStyle(
    fontSize: 16,
    fontWeight: FontWeight.w400,
    height: 1.5,
    fontFamily: fontFamily,
  );
  static const bodyMedium = TextStyle(
    fontSize: 14,
    fontWeight: FontWeight.w400,
    height: 1.5,
    fontFamily: fontFamily,
  );
  static const bodySmall = TextStyle(
    fontSize: 12,
    fontWeight: FontWeight.w400,
    height: 1.5,
    fontFamily: fontFamily,
  );

  static const labelLarge = TextStyle(
    fontSize: 14,
    fontWeight: FontWeight.w600,
    height: 1.4,
    letterSpacing: 0.5,
    fontFamily: fontFamily,
  );
  static const labelMedium = TextStyle(
    fontSize: 12,
    fontWeight: FontWeight.w500,
    height: 1.4,
    letterSpacing: 0.5,
    fontFamily: fontFamily,
  );
  static const labelSmall = TextStyle(
    fontSize: 10,
    fontWeight: FontWeight.w500,
    height: 1.4,
    letterSpacing: 0.8,
    fontFamily: fontFamily,
  );

  static const monoMedium = TextStyle(
    fontSize: 14,
    fontWeight: FontWeight.w400,
    height: 1.6,
    fontFamily: fontFamilyMono,
  );
}

File: app/JOI/Core/Theme/JOITypography.swift
import SwiftUI

enum JOITypography {
    static let fontFamily = "Inter"
    static let fontFamilyMono = "JetBrains Mono"

#if os(iOS)
    private static let bodyLargeSize: CGFloat = 17
    private static let bodyMediumSize: CGFloat = 15
    private static let bodySmallSize: CGFloat = 13
    private static let labelLargeSize: CGFloat = 15
    private static let labelMediumSize: CGFloat = 13
    private static let labelSmallSize: CGFloat = 11
#else
    private static let bodyLargeSize: CGFloat = 16
    private static let bodyMediumSize: CGFloat = 14
    private static let bodySmallSize: CGFloat = 12
    private static let labelLargeSize: CGFloat = 14
    private static let labelMediumSize: CGFloat = 12
    private static let labelSmallSize: CGFloat = 10
#endif

    static let headlineLarge = Font.system(size: 28, weight: .semibold)
    static let headlineMedium = Font.system(size: 22, weight: .semibold)
    static let headlineSmall = Font.system(size: 18, weight: .semibold)

    static let bodyLarge = Font.system(size: bodyLargeSize, weight: .regular)
    static let bodyMedium = Font.system(size: bodyMediumSize, weight: .regular)
    static let bodySmall = Font.system(size: bodySmallSize, weight: .regular)

    static let labelLarge = Font.system(size: labelLargeSize, weight: .semibold)
    static let labelMedium = Font.system(size: labelMediumSize, weight: .medium)
    static let labelSmall = Font.system(size: labelSmallSize, weight: .medium)

    static let monoMedium = Font.system(size: bodyMediumSize, weight: .regular, design: .monospaced)
    static let monoSmall = Font.system(size: bodySmallSize, weight: .regular, design: .monospaced)
}
